Packages/ClientRuntime/Sources/Middleware/OperationStack.swift

```diff
@@ -7,76 +7,65 @@ public struct OperationStack<OperationStackInput: Encodable & Reflection,
+    public var deserializeStep: DeserializeStep<OperationStackOutput>
     
     public init(id: String) {
         self.id = id
         self.initializeStep = InitializeStep<OperationStackInput,
-                                             OperationStackOutput,
-                                             OperationStackError>(id: InitializeStepId)
+                                             OperationStackOutput>(id: InitializeStepId)
         self.serializeStep = SerializeStep<OperationStackInput,
-                                           OperationStackOutput,
-                                           OperationStackError>(id: SerializeStepId)
-        self.buildStep = BuildStep<OperationStackOutput,
-                                   OperationStackError>(id: BuildStepId)
-        self.finalizeStep = FinalizeStep<OperationStackOutput,
-                                         OperationStackError>(id: FinalizeStepId)
-        self.deserializeStep = DeserializeStep<OperationStackOutput,
-                                               OperationStackError>(id: DeserializeStepId)
+                                           OperationStackOutput>(id: SerializeStepId)
+        self.buildStep = BuildStep<OperationStackOutput>(id: BuildStepId)
+        self.finalizeStep = FinalizeStep<OperationStackOutput>(id: FinalizeStepId)
+        self.deserializeStep = DeserializeStep<OperationStackOutput>(id: DeserializeStepId)
         
     }
     
     /// This execute will execute the stack and use your next as the last closure in the chain
     public func handleMiddleware<H: Handler>(context: HttpContext,
                                              input: OperationStackInput,
-                                             next: H) -> SdkResult<OperationStackOutput, OperationStackError>
+                                             next: H) async throws -> OperationStackOutput
     where H.Input == SdkHttpRequest,
           H.Output == OperationOutput<OperationStackOutput>,
-          H.Context == HttpContext,
-          H.MiddlewareError == SdkError<OperationStackError> {
+          H.Context == HttpContext {
         
         let deserialize = compose(next: DeserializeStepHandler(handler: next), with: deserializeStep)
         let finalize = compose(next: FinalizeStepHandler(handler: deserialize), with: finalizeStep)
         let build = compose(next: BuildStepHandler(handler: finalize), with: buildStep)
         let serialize = compose(next: SerializeStepHandler(handler: build), with: serializeStep)
         let initialize = compose(next: InitializeStepHandler(handler: serialize), with: initializeStep)
         
-        let result = initialize.handle(context: context, input: input)
-        return result.flatMap { operationOutput in
-            return .success(operationOutput.output!)
-        }
+        let result = try await initialize.handle(context: context, input: input)
+        return result.output!
```

Comment:
Now that we have changed the function signature to include  `throw`, should we do something like:

```
let result = try await initialize.handle(...)
guard let output = result.output else {
   throw ....
}
return output
```
??

Reply:
updated